### DISCUSSION TITLE
Show live tabs only

### DISCUSSION THREAD

**Discussion by KogasaPls** [Chatterino/chatterino2#3656] [Q&A] [answered]
I have a large number of tabs, the majority of which are usually offline. So I wrote a quick patch to hide the offline ones, which can be found on GitHub [here.](https://github.com/KogasaPls/chatterino2/tree/hide-offline) Here's a quick [video demo](https://streamable.com/vvby9y) as well. This is purely a visual change (offline chats are still connected) unlike what is requested in #2334. 

Before submitting a pull request, I would like to gauge interest and ask what I need to add/change for a PR to be viable. I would appreciate the feedback, even if it is to say that there is no interest in this feature being developed.

Thanks!

**Comment by Felanbird** [comment]
> Before submitting a pull request, I would like to gauge interest and ask what I need to add/change for a PR to be viable. 

That's basically the point of a PR, you submit code and we review it, even if the answer is no you've already done the work so you might as well get some form of review.

Immediate comments without testing or looking at the code:
1. Will this naturally drop channels that go offline and re-arrange ones that go online?
2. This should not have a default hotkey, we already have enough problems with people accidentally pressing `ctrl+u`, but - it should have the functionality to add a hotkey for it via our custom hotkeys.

**Reply by Felanbird** [reply]
![image](https://user-images.githubusercontent.com/41973452/162134957-9bd978cb-bb69-4541-b1e9-b9ab814afddb.png)

**Reply by KogasaPls** [reply]
Thanks, I'll go ahead and submit a PR with the default hotkey removed. The answer to 1. is yes. When enabled, it runs performLayout() and rebuilds the "live tabs" list whenever a tab updates `isLive_`. New channels which go live are inserted into their proper place, and channels which go offline disappear.

**Comment by pajlada** [comment, marked as answer]
The latest nightly version has this this functionality where you can configure Chatterino to only show tabs where one of the chats in it has a stream that is online.

Implemented in https://github.com/Chatterino/chatterino2/pull/4358

Thank you for the suggestion and the initial work :heart:
